fastlane_bot/helpers/carbon_trade_splitter.py

```diff
@@ -0,0 +1,195 @@
+import json
+from typing import List, Dict
+
+from fastlane_bot.helpers import TradeInstruction
+
+
+class CarbonTradeSplitter:
+    NATIVE = "native"
+    WRAPPED = "wrapped"
+    NEITHER = "neither"
+
+    def __init__(self, ConfigObj):
+        self.ConfigObj = ConfigObj
+
+    def split_carbon_trades(
+        self, trade_instructions: List[TradeInstruction]
+    ) -> List[TradeInstruction]:
+        """Splits Carbon trades that cannot be aggregated into a single trade action.
+
+        This function split a single Carbon trade into multiple trades for cases that include a mix of tokens or different Carbon deployments.
+        For example NATIVE/WRAPPED -> TKN -> NATIVE -> TKN & WRAPPED -> TKN.
+
+        Args:
+            trade_instructions: The list of TradeInstruction objects.
+
+        Returns:
+            The processed list of TradeInstruction objects after splitting incompatible trades.
+
+        """
+        new_trade_list = []
+        for trade in trade_instructions:
+            if not self._is_carbon_trade(trade):
+                new_trade_list.append(trade)
+                continue
+
+            carbon_exchanges = self._process_carbon_trades(trade)
+            self._create_new_trades_from_carbon_exchanges(
+                carbon_exchanges, trade, new_trade_list
+            )
```

Comment:
This function (`split_carbon_trades`) is in charge of returning a list of trades.
So my suggestion is that it will be the only one populating the `new_trade_list` array which it returns.
In other words, function `_create_new_trades_from_carbon_exchanges` should return an independent list of trades, and this function should concatenate that list to its (larger) list.
This function may as well call function `_process_carbon_trades` internally (or perhaps even just merge both of these functions into a single function).